net-print/cnijfilter-drivers/cnijfilter-drivers-3.80.ebuild

```diff
@@ -0,0 +1,148 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit autotools multibuild
+
+MY_P="cnijfilter-source-${PV}-1"
+
+DESCRIPTION="CUPS drivers for some Canon inkjet printers (printer specific part)"
+HOMEPAGE="http://support-sg.canon-asia.com/contents/SG/EN/0100469302.html"
+SRC_URI="http://gdlp01.c-wss.com/gds/3/0100004693/01/${MY_P}.tar.gz"
+
+LICENSE="GPL-2"
+
+# each source version supports a set of printer models
+# that is not supported in other versions
+# so slot each version to have all printer models
+
+SLOT="${PV}"
+
+KEYWORDS="~amd64 ~x86"
+PRINTERS=(401:mp230 402:mg2200 403:e510 404:mg3200 405:mg4200 406:ip7200 407:mg5400 408:mg6300)
+IUSE="${PRINTERS[@]#*:} +servicetools"
+REQUIRED_USE="|| ( ${PRINTERS[@]#*:} )"
+
+RDEPEND="
+	>=net-print/cnijfilter-3.80[servicetools?]
+	>=media-libs/libpng-1.5:0=
+	>=media-libs/tiff-3.4:0=
+	>=net-print/cups-1.4
+	servicetools? (
+		>=dev-libs/libxml2-2.7.3-r2
+		>=x11-libs/gtk+-2.6:2
+	)
+"
+DEPEND="${RDEPEND}
+	sys-devel/gettext
+"
+
+S="${WORKDIR}/${MY_P}"
+
+PATCHES=(
+	"${FILESDIR}/cnijfilter-3.70-png.patch"
+	"${FILESDIR}/cnijfilter-3.70-ppd.patch"
+	"${FILESDIR}/cnijfilter-3.70-ppd2.patch"
+	"${FILESDIR}/cnijfilter-3.70-libexec-cups.patch"
+	"${FILESDIR}/cnijfilter-3.70-libexec-backend.patch"
+	"${FILESDIR}/cnijfilter-3.80-cups.patch"
+	)
+
+pkg_setup() {
+	[[ -z ${LINGUAS} ]] && LINGUAS="en"
+
+	if [[ ${MERGE_TYPE} == source ]]; then
+
+		# select components
+
+		DIRS=(cnijfilter)
+		# lgmon must be first as cngpijmon depends on it
+		use servicetools && DIRS+=(lgmon cngpijmon maintenance)
+
+		# select variants
+
+		for (( i=0; i<${#PRINTERS[@]}; i++ )); do
```

Comment:
The whole point was that you don't have to do the `(( ... ))` anymore and you can just do `for x in "${PRINTERS[@]}"`

Reply:
Fixed